### PR TITLE
[new package] add rio terminal

### PR DESCRIPTION
**Pull request by Kreijstal**
Fixed the LTO issue on upstream!

### CODE REVIEW

File: mingw-w64-rio/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+pkgdesc="A hardware-accelerated GPU terminal emulator powered by WebGPU (mingw-w64)"
+arch=('any')
+mingw_arch=('mingw64' 'ucrt64' 'clang64')
+url="https://github.com/raphamorim/rio"
+license=('spdx:MIT')
+makedepends=("${MINGW_PACKAGE_PREFIX}-rust"
+             "${MINGW_PACKAGE_PREFIX}-cc"
+             "${MINGW_PACKAGE_PREFIX}-pkgconf"
+             "${MINGW_PACKAGE_PREFIX}-cmake"
+             "${MINGW_PACKAGE_PREFIX}-freetype"
```
Comment (Biswa96) [inline mingw-w64-rio/PKGBUILD:15]: Are you sure these dependencies are required in this build?

File: mingw-w64-rio/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+prepare() {
+  cd "${_realname}-${pkgver}" 
+
+  cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
+}
+
+build() {
+  cd "${_realname}-${pkgver}"
+
+  cargo build --frozen --release --all-features
```
Comment (ognevny) [inline mingw-w64-rio/PKGBUILD:30]: is `--all-features` required? I see all useful features are already default upstream

File: mingw-w64-rio/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,38 @@
+_realname=rio
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=0.1.15
+pkgrel=1
+pkgdesc="A hardware-accelerated GPU terminal emulator powered by WebGPU (mingw-w64)"
+arch=('any')
+mingw_arch=('mingw64' 'ucrt64' 'clang64')
```
Comment (Biswa96) [inline mingw-w64-rio/PKGBUILD:8]:
```suggestion
mingw_arch=('ucrt64' 'clang64' 'clangarm64')
```

File: mingw-w64-rio/PKGBUILD (hunk 1 above)
Comment (ognevny) [reply mingw-w64-rio/PKGBUILD:15]: rust only can be left. cmake is usually used for libz-ng-sys, which is not used for package, freetype and fontconfig are probably used for Linux/macos, I didn't find them all in build logs

**Comment by Biswa96** [comment]
rio package has been added from #22248

**Pull request closed without merging**